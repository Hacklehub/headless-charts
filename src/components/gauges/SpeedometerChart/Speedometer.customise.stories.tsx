import { Meta, StoryObj } from '@storybook/react';

import SpeedometerChart from '.';

const meta: Meta<typeof SpeedometerChart> = {
  title: 'Gauge/Speedometer/customized',
  component: SpeedometerChart,
  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof SpeedometerChart>;

export const WithRegions: Story = {
  args: {
    data: 0.7,
    label: {
      text: 'Coverage',
    },
    id: 'speedometer-with-regions',
    regions: [
      {
        limit: 0.5,
        className: 'fill-red-500',
      },
      {
        limit: 0.8,
        className: 'fill-yellow-500',
      },
      {
        limit: 0.9,
        className: 'fill-green-500',
      },
      {
        limit: 1,
        className: 'fill-blue-500',
      },
    ],
  },
};
