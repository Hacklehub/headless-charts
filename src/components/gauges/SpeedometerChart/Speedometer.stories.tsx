import { Meta } from '@storybook/react';
import SpeedometerChart from '.';

export default {
  title: 'Gauge/Speedometer',
  component: SpeedometerChart,
  tags: ['autodocs'],
} as Meta;

export const Default = {
  args: {
    data: 0.7,
    label: {
      text: 'Coverage',
    },
    id: 'speedometer-default',
  },
};

export const WithRadius = {
  args: {
    ...Default.args,
    id: 'speedometer-with-radius',
    needleRadius: 0.5,
  },
};

export const WithAxisTicks = {
  args: {
    ...Default.args,
    id: 'speedometer-with-axis-ticks',
    axisTicks: 10,
  },
};

export const WithRegions = {
  args: {
    ...Default.args,
    id: 'speedometer-with-regions',
    regions: [
      {
        limit: 0.5,
        className: 'fill-red-500',
      },
      {
        limit: 0.8,
        className: 'fill-yellow-500',
      },
      {
        limit: 0.9,
        className: 'fill-green-500',
      },
      {
        limit: 1,
        className: 'fill-blue-500',
      },
    ],
  },
};


