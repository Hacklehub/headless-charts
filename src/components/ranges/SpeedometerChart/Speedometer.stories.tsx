import { Meta } from '@storybook/react';
import SpeedometerChart from '.';

export default {
  title: 'ranges/Speedometer',
  component: SpeedometerChart,
  tags: ['autodocs'],
} as Meta;

export const Default = {
  args: {
    data: 0.7,
    label: {
      text: 'Coverage',
    },
    id: 'speedometer-default',
  },
};

export const WithRadius = {
  args: {
    ...Default.args,
    id: 'speedometer-with-radius',
    needleRadius: 0.5,
  },
};

export const WithAxisTicks = {
  args: {
    ...Default.args,
    id: 'speedometer-with-axis-ticks',
    axisTicks: 10,
  },
};

export const WithRegions = {
  args: {
    ...Default.args,
    id: 'speedometer-with-regions',
    regions: [
      {
        limit: 0,
      },
      {
        limit: 1,
      },
      {
        limit: 2,
      },
      {
        limit: 3,
      },
    ],
  },
};
