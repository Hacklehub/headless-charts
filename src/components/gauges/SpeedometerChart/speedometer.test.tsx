import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';

import SpeedometerChart from '.';

describe('SpeedometerChart', () => {
  it('renders speedometer component for snapshot test', () => {
    render(<SpeedometerChart data={50} id='speedometer-chart' />);
    const speedometer = screen.getByTestId('speedometer');
    expect(speedometer).toMatchSnapshot();
  });

  it('renders speedometer component', () => {
    render(<SpeedometerChart data={50} id='speedometer-chart' />);
    const speedometer = screen.getByTestId('speedometer');
    expect(speedometer).toBeInTheDocument();
  });

})
