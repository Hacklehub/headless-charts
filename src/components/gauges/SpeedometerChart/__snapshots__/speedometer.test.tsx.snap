// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpeedometerChart > renders speedometer component 1`] = `
<svg
  class="chart dark:bg-gray-800 dark:text-gray-50 h-64 lg:w-4/12 md:w-6/12 text-gray-900 w-full"
  data-testid="speedometer"
  id="speedometer-chart"
>
  <g
    text-anchor="middle"
    transform="translate(0,0)"
  >
    <g
      class="gauge-levels"
    />
    <g
      class="data-group"
      transform="rotate(0)"
    >
      <circle
        class="fill-current stroke-current"
        cx="0"
        cy="0"
        r="3"
      />
      <line
        class="fill-current stroke-current stroke-2"
        x1="0"
        x2="32"
        y1="0"
        y2="0"
      />
      <path
        class="fill-current stroke-current stroke-2"
        d="M0,-6.204L5.373,3.102L-5.373,3.102Z"
        transform="translate(32,0)rotate(-90)"
      />
    </g>
    <g
      transform="rotate(-144) translate(-28,0)"
    >
      <line
        class="axis-line stroke-current"
        x1="4"
        x2="-4"
      />
      <text
        class="text-xs fill-current"
        transform="rotate(90)translate(0,16)"
      >
        0.2
      </text>
    </g>
    <g
      transform="rotate(-108) translate(-28,0)"
    >
      <line
        class="axis-line stroke-current"
        x1="4"
        x2="-4"
      />
      <text
        class="text-xs fill-current"
        transform="rotate(90)translate(0,16)"
      >
        0.4
      </text>
    </g>
    <g
      transform="rotate(-72) translate(-28,0)"
    >
      <line
        class="axis-line stroke-current"
        x1="4"
        x2="-4"
      />
      <text
        class="text-xs fill-current"
        transform="rotate(90)translate(0,16)"
      >
        0.6
      </text>
    </g>
    <g
      transform="rotate(-36) translate(-28,0)"
    >
      <line
        class="axis-line stroke-current"
        x1="4"
        x2="-4"
      />
      <text
        class="text-xs fill-current"
        transform="rotate(90)translate(0,16)"
      >
        0.8
      </text>
    </g>
    <g
      transform="rotate(0) translate(-28,0)"
    >
      <line
        class="axis-line stroke-current"
        x1="4"
        x2="-4"
      />
      <text
        class="text-xs fill-current"
        transform="rotate(90)translate(0,16)"
      >
        1
      </text>
    </g>
  </g>
</svg>
`;
